charter.md

```diff
@@ -2,24 +2,45 @@
+Recognizing the need for a standardized security protocol to support group key establishment, authentication, and confidentiality services for messaging, the IETF has specified the Messaging Layer Security (MLS) protocol [I-D.ietf-mls-protocol] and architecture [I-D.ietf-mls-architecture]. MLS is agnostic to the identity system used within any given messaging service; it provides confidentiality of sessions once the participants in a conversation have been identified. To achieve interoperable messaging, the MIMI working group will identify profiles of one or more existing identity systems suitable for establishing end-to-end cryptographic identity across messaging services, assuming the use of MLS for key establishment. 
 
-Achieving interoperability requires support for user discovery, i.e., the ability for a sending user in one application to find a recipient user in a different application. The working group will specify one or more mechanisms to enable user discovery, together with best practice recommendations for functionality, configuration options, and other aspects that may require implementation outside the messaging applications themselves (e.g., in operating systems). At a minimum, user discovery based on phone numbers and email addresses must both be specified. New means of user discovery must support common user control and privacy features (e.g., allowing users to block contacts, control their own discoverability, etc.). Express user preferences about discoverability must be respected. User discovery mechanisms should also be designed to minimize the accumulation of user contact or relationship data by intermediaries.
+Achieving interoperability requires support for user discovery, i.e., the ability for a sending user in one application to find a recipient user in a different application. The working group will specify one or more mechanisms to enable user discovery, together with best practice recommendations for functionality, configuration options, and other aspects. At a minimum, user discovery within a single service, based on Internet domain-based addresses (ex: email addresses), and phone numbers should both be specified. New means of user discovery must support common user control and privacy features (e.g., allowing users to block contacts, control their own discoverability, etc.). Express user preferences about discoverability must be respected. User discovery mechanisms should also be designed to minimize the accumulation of user contact or relationship data by intermediaries.
```

Comment:
What do we gain by saying "Internet domain-based addresses (ex: email addresses)"? From the user perspective this is the identifier they type in to find another user, i.e., an email address.

Reply:
My personal email address (rohan.mahy@gmail.com) and my personal wire handle (@rohan_mahy@wire.com) are very different. They both have an @ sign and domain.  Technically the only thing we should care about is that it has an @ sign and a domain. If the service you are querying has your email address and you allow account to be found via email address, that should work. If you have a separate handle and someone searches for you by handle and you authorized that, it should find you that way. As with the phone number case, you need to know which service you are searching, otherwise you are in a world of privacy violations and spam.

Reply:
I’m not super convinced by this for charter purposes. It seems to assume a lot about the ultimate design of the discovery service. In some way, look-ups of phone numbers and email addresses need to be supported, whether using an overlay service or a service that proves each app individually. Having support for other kinds of handles would make the user experience better, and may be required for some apps, but I don’t think it is strictly required for interop among all apps.